### PR TITLE
[Chat][Feature] Chat UI infrastructure

### PR DESCRIPTION
**Pull request by Mohtasim**
## Purpose

* Upgrade Kotlin Extentsion Version
* Add compose dependencies into chat project
* Adding ChatView in chatActivity
* Adding blank FluentUI theme in compose view

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull Request Checklist

- [ ] Public API changes
- [ ] Verified for cross-platform
- [ ] Synced with iOS, Web team
- [ ] Internal review completed
- [x] UI Changes
  - [ ] Screen captures included
  - [ ] Tested for Light/Dark mode
  - [ ] Tested for screen rotation
  - [ ] Tested on Tablet and small screen device (5")
  - [ ] Include localization changes
- [ ] Tests
  - [ ] Memory leak analysis performed
  - [ ] Tested on API 21, 26 and latest 
  - [ ] Tested for background mode
  - [ ] Unit Tests Included
  - [ ] UI Automated Tests Included

## How to Test

* Open...
* Click on...

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: azure-communication-ui/chat/build.gradle [hunk 1]
```diff
@@ -47,13 +47,22 @@ dependencies {
...
     api ("com.azure.android:azure-communication-common:$azure_common_sdk_version")
 
+    implementation "com.microsoft.fluentui:fluentui_others:$microsoft_fluent_ui_version_v2"
+    implementation "androidx.compose.ui:ui:$compose_version"
+    implementation "androidx.compose.material:material:$compose_version"
+    implementation "androidx.compose.ui:ui-tooling-preview:$compose_version"
+    implementation "androidx.activity:activity-compose:$androidx_activity_compose_version"
+    implementation "androidx.lifecycle:lifecycle-runtime-ktx:$androidx_lifecycle_runtime_ktx_version"
+    implementation "androidx.compose.material3:material3:$compose_material3_version"
+    implementation "androidx.compose.material3:material3-window-size-class:$compose_material3_version"
```
Comment (Mrayyan) [inline azure-communication-ui/chat/build.gradle:60]: I don't think we need any material dependencies.

Comment (Mohtasim) [reply azure-communication-ui/chat/build.gradle:60]: Added for the custom topAppbarDeafaults. 

import androidx.compose.material.MaterialTheme
import androidx.compose.material.Text
import androidx.compose.material3.CenterAlignedTopAppBar
import androidx.compose.material3.TopAppBarDefaults

This is used in the topbar of the chatView. But I agree if we build this components ourself, we can remove such dependencies.

**Review by ShaunaSong — DISMISSED** [review]

**Review by ShaunaSong — DISMISSED** [review]

File: azure-communication-ui/build.gradle [hunk 2]
```diff
@@ -51,11 +51,16 @@ buildscript {
             return rootProject.hasProperty("disableTaskRootCheck") && rootProject.getProperty("disableTaskRootCheck")
         }
 
-        compose_version = '1.3.1'
+        androidx_lifecycle_runtime_ktx_version = '2.3.1'
+        androidx_activity_compose_version = '1.5.1'
```
Comment (iaulakh) [inline azure-communication-ui/build.gradle:55]: @Mohtasim I think these are included for calling too? are we creating separate version for Chat? I think we will use same version.

File: azure-communication-ui/build.gradle [hunk 3]
```diff
@@ -51,11 +51,16 @@ buildscript {
...
+        compose_version = '1.2.1'
+        compose_material3_version = '1.0.0-alpha12'
```
Comment (iaulakh) [inline azure-communication-ui/build.gradle:57]: for which component we are using material version? I think we will add if required in futre.

File: azure-communication-ui/build.gradle [hunk 4]
```diff
@@ -51,11 +51,16 @@ buildscript {
...
+        kotlin_compiler_extension_version = '1.3.1'
     }
 
     repositories {
         google()
+        mavenCentral()
```
Comment (iaulakh) [inline azure-communication-ui/build.gradle:63]: would it be possible to know the cause for adding maven here?

File: azure-communication-ui/chat/src/main/res/values/azure_communication_ui_chat_strings.xml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?><!--
+   Copyright (c) Microsoft Corporation. All rights reserved.
+   Licensed under the MIT License.
+  -->
+
+<resources>
+    <string name="azure_communication_ui_chat_title_activity_compose_chat">ComposeChatActivity</string>
+    <string name="azure_communication_ui_chat_action_bar_title">Chat</string>
```
Comment (iaulakh) [inline azure-communication-ui/chat/src/main/res/values/azure_communication_ui_chat_strings.xml:8]: are we going to use these strings in Chat UI? we need only for demo app no Ui library

File: azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/presentation/components/ChatCompsiteActionBar.kt [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+import androidx.compose.ui.graphics.drawscope.clipRect
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.core.graphics.toColorInt
+import com.azure.android.communication.ui.chat.R
+import com.azure.android.communication.ui.chat.presentation.components.microcomponents.ChatCompositeActionBarBackButton
+
+@Composable
+fun ChatCompsiteActionBar(
```
Comment (iaulakh) [inline azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/presentation/components/ChatCompsiteActionBar.kt:31]: We will checkin the action bar required as per figma with actual colors. We do not need to checkin temp code into develop.

File: azure-communication-ui/chat/build.gradle [hunk 7]
```diff
@@ -47,13 +47,24 @@ dependencies {
...
     testImplementation "junit:junit:$junit_version"
     androidTestImplementation "androidx.test.ext:junit:$androidx_junit_version"
     androidTestImplementation "androidx.test.espresso:espresso-core:$androidx_espresso_core_version"
+    androidTestImplementation "androidx.compose.ui:ui-test-junit4:$compose_version"
+    debugImplementation "androidx.compose.ui:ui-tooling:$compose_version"
+    debugImplementation "androidx.compose.ui:ui-tooling-data:$compose_version"
+    debugImplementation "androidx.compose.ui:ui-test-manifest:$compose_version"
+
+    debugImplementation "androidx.customview:customview:1.2.0-alpha01"
+    debugImplementation "androidx.customview:customview-poolingcontainer:1.0.0-alpha01"
```
Comment (iaulakh) [inline azure-communication-ui/chat/build.gradle:69]: Please add a comment here for fix

File: azure-communication-ui/chat/build.gradle [hunk 8]
```diff
@@ -47,13 +47,24 @@ dependencies {
...
-    implementation "androidx.activity:activity-compose:$androidx_activity_compose_version"
-
     api ("com.azure.android:azure-communication-chat:$azure_chat_sdk_version")
     api ("com.azure.android:azure-communication-common:$azure_common_sdk_version")
 
+    implementation "com.microsoft.fluentui:fluentui_others:$microsoft_fluent_ui_version_v2"
+    implementation "androidx.compose.ui:ui:$compose_version"
+    implementation "androidx.compose.material:material:$compose_version"
+    implementation "androidx.compose.ui:ui-tooling-preview:$compose_version"
+    implementation "androidx.activity:activity-compose:1.5.1"
```
Comment (iaulakh) [inline azure-communication-ui/chat/build.gradle:57]: 1.5.1 can be defined as version.

File: azure-communication-ui/chat/build.gradle [hunk 9]
```diff
@@ -47,13 +47,24 @@ dependencies {
...
-    implementation "com.microsoft.fluentui:fluentui_others:$microsoft_fluent_ui_version_v2"
-    implementation "androidx.activity:activity-compose:$androidx_activity_compose_version"
-
     api ("com.azure.android:azure-communication-chat:$azure_chat_sdk_version")
     api ("com.azure.android:azure-communication-common:$azure_common_sdk_version")
 
+    implementation "com.microsoft.fluentui:fluentui_others:$microsoft_fluent_ui_version_v2"
+    implementation "androidx.compose.ui:ui:$compose_version"
+    implementation "androidx.compose.material:material:$compose_version"
+    implementation "androidx.compose.ui:ui-tooling-preview:$compose_version"
```
Comment (iaulakh) [inline azure-communication-ui/chat/build.gradle:56]: do we need to add material depepndency? I think we can avoid as per discussion with Adam

File: azure-communication-ui/chat/build.gradle (hunk 1 above)
Comment (Mohtasim) [reply azure-communication-ui/chat/build.gradle]: Removed the material3 dependency.

File: azure-communication-ui/build.gradle (hunk 2 above)
Comment (Mohtasim) [reply azure-communication-ui/build.gradle:55]: This is newly added. Calling uses a different one. $androidx_lifecycle_viewmodel_ktx_version

File: azure-communication-ui/build.gradle (hunk 3 above)
Comment (Mohtasim) [reply azure-communication-ui/build.gradle:57]: Sorry. Removed this from dependency but did not remove from version gradle script. Removing this.

File: azure-communication-ui/chat/src/main/res/values/azure_communication_ui_chat_strings.xml (hunk 5 above)
Comment (Mohtasim) [reply azure-communication-ui/chat/src/main/res/values/azure_communication_ui_chat_strings.xml:8]: For ChatView's topbar, we will need this string according to figma as well.

File: azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/presentation/components/ChatCompsiteActionBar.kt (hunk 6 above)
Comment (Mohtasim) [reply azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/presentation/components/ChatCompsiteActionBar.kt:31]: Checked in a different basic action bar component UI to demonstrate an example of adding any component.

File: azure-communication-ui/build.gradle (hunk 4 above)
Comment (Mohtasim) [reply azure-communication-ui/build.gradle:63]: This was added by mistake. Mixed up this repo block with allrepo block. Removing this in next commit.

File: azure-communication-ui/chat/build.gradle (hunk 7 above)
Comment (Mohtasim) [reply azure-communication-ui/chat/build.gradle:69]: Thank you. adding the comment.

File: azure-communication-ui/chat/build.gradle (hunk 9 above)
Comment (Mohtasim) [reply azure-communication-ui/chat/build.gradle:56]: Removing the material in next commit. Thanks

File: azure-communication-ui/chat/src/main/res/values/azure_communication_ui_chat_strings.xml (hunk 5 above)
Comment (Mohtasim) [reply azure-communication-ui/chat/src/main/res/values/azure_communication_ui_chat_strings.xml:8]: Removed the strings.

**Review by iaulakh — CHANGES_REQUESTED** [review]

**Review by iaulakh — APPROVED** [review]

**Review by ShaunaSong — APPROVED** [review]

**Pull request merged**